plugins/woocommerce/includes/data-stores/class-wc-customer-data-store-session.php

```diff
@@ -147,8 +145,8 @@ public function read( &$customer ) {
 				}
 				if ( ! empty( $data[ $session_key ] ) && is_callable( array( $customer, "set_{$function_key}" ) ) ) {
 					if ( 'meta_data' === $session_key ) {
-						$meta_data_values = maybe_unserialize( $data[ $session_key ] );
-						if ( $meta_data_values ) {
+						$meta_data_values = is_serialized( $data[ $session_key ], true ) ? maybe_unserialize( $data[ $session_key ] ) : $data[ $session_key ];
```

Comment:
I see `maybe_unserialize` calls `is_serialized` with strict mode `true` anyway, so is the change on this line needed?

I see this line doing basically the same thing as a direct call to `maybe_unserialize`. Line below is definitely needed though.

Reply:
I think we can skip this all together, this is intended for Additional Fields with individual fields which didn't ship yet, so we can just check if it's an array.

If it was serialized before, it will be overridden after first load.